allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_nwb_api.py

```diff
@@ -311,3 +327,171 @@ def get_motion_correction(self) -> pd.DataFrame:
+        """
+        Gets corneal, eye, and pupil ellipse fit data
+
+        Returns
+        -------
+        pd.DataFrame
+            *_center_x
+            *_center_y
+            *_height
+            *_phi
+            *_width
+
+            where "*" can be "corneal", "pupil" or "eye"
+
+        or None if no eye tracking data
+        """
+        try:
+            et_mod = self.nwbfile.get_processing_module("eye_tracking")
+            rgm_mod = self.nwbfile.get_processing_module("raw_gaze_mapping")
+        except KeyError as e:
+            warnings.warn(f"This ophys session '{int(self.nwbfile.identifier)}' has no eye tracking data. (NWB error: {e})")
+            return None
+
+        cr_ellipse_fits = et_mod.get_data_interface("cr_ellipse_fits").to_dataframe()
+        eye_ellipse_fits = et_mod.get_data_interface("eye_ellipse_fits").to_dataframe()
+        pupil_ellipse_fits = et_mod.get_data_interface("pupil_ellipse_fits").to_dataframe()
+
+        # NOTE: ellipse fit "height" and "width" parameters describe the
+        # "half-height" and "half-width" of fitted ellipse.
+        eye_tracking_data = {
+            "corneal_reflection_center_x": cr_ellipse_fits["center_x"].values,
+            "corneal_reflection_center_y": cr_ellipse_fits["center_y"].values,
+            "corneal_reflection_height": 2 * cr_ellipse_fits["height"].values,
+            "corneal_reflection_width": 2 * cr_ellipse_fits["width"].values,
+            "corneal_reflection_phi": cr_ellipse_fits["phi"].values,
+
+            "pupil_center_x": pupil_ellipse_fits["center_x"].values,
+            "pupil_center_y": pupil_ellipse_fits["center_y"].values,
+            "pupil_height": 2 * pupil_ellipse_fits["height"].values,
+            "pupil_width": 2 * pupil_ellipse_fits["width"].values,
+            "pupil_phi": pupil_ellipse_fits["phi"].values,
+
+            "eye_center_x": eye_ellipse_fits["center_x"].values,
+            "eye_center_y": eye_ellipse_fits["center_y"].values,
+            "eye_height": 2 * eye_ellipse_fits["height"].values,
+            "eye_width": 2 * eye_ellipse_fits["width"].values,
+            "eye_phi": eye_ellipse_fits["phi"].values
+        }
+
+        timestamps = rgm_mod.get_data_interface("eye_area").timestamps[:]
+        index = pd.Index(data=timestamps, name="Time (s)")
+        return pd.DataFrame(eye_tracking_data, index=index)
+
+    @staticmethod
+    def _add_eye_tracking_data_to_nwb(nwbfile: NWBFile,
+                                      eye_dlc_ellipses_path: str,
+                                      eye_tracking_rig_geometry: dict,
+                                      eye_gaze_mapping_file_path: str,
+                                      session_sync_file: Path):
+        nwb.add_eye_tracking_rig_geometry_data_to_nwbfile(nwbfile=nwbfile,
+                                                          eye_tracking_rig_geometry=eye_tracking_rig_geometry)
+
+        # Collect eye tracking/gaze mapping data from files
```

Comment:
I'm not sure lines 485-492 are doing the right thing. It looks like this is saving the eye tracking data the way it's saved for the Ecephys Project, but the way that the visual behavior project would like to save eye tracking data is via the output (`pd.DataFrame`) of https://github.com/AllenInstitute/AllenSDK/blob/43c4fa2557689b9b1029c2d7f1efdc7c684a651d/allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_ophys_data_xforms.py#L398.

Reply:
This is related to:
https://github.com/AllenInstitute/AllenSDK/pull/1815/files#r561168388

Reply:
OK, I was confused by a comment you made earlier that processing is not needed for the eye tracking data (which is what the method you linked to does) so I didn't use it. I will go ahead and change the ellipse logic to use the `get_eye_tracking` method

Reply:
I have updated to use the dataframe output from `get_eye_tracking`